### PR TITLE
UGENE-6799 Task member becomes invalid during the task execution

### PR DESCRIPTION
**Pull request by DmitriiSukhomlinov**

### CODE REVIEW

File: src/plugins/workflow_designer/src/WorkflowSceneIOTasks.cpp [hunk 1]
```diff
@@ -101,6 +101,7 @@ Task::ReportResult LoadWorkflowSceneTask::report() {
     if (hasError()) {
         return ReportResult_Finished;
     }
+    CHECK(scene != nullptr, ReportResult_Finished);
```
Comment (atiunov) [inline src/plugins/workflow_designer/src/WorkflowSceneIOTasks.cpp:104]: "scene" pointer should be QPointer to be nullified on the WD closing.

File: src/plugins/workflow_designer/src/WorkflowViewController.cpp [hunk 2]
```diff
@@ -304,6 +304,12 @@ WorkflowView::WorkflowView(WorkflowGObject *go)
 
 WorkflowView::~WorkflowView() {
     uiLog.trace("~WorkflowView");
+    if (!loadWorkflowSceneTask.isNull()) {
+        TaskScheduler*  ts = AppContext::getTaskScheduler();
+        SAFE_POINT(ts != nullptr, "TaskScheduler isn't found", );
+
+        ts->cancelTask(loadWorkflowSceneTask);
```
Comment (atiunov) [inline src/plugins/workflow_designer/src/WorkflowViewController.cpp]: You can just use Task::cancel() here.

**Review by atiunov — CHANGES_REQUESTED** [review]

File: src/plugins/workflow_designer/src/WorkflowSceneIOTasks.cpp (hunk 1 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/workflow_designer/src/WorkflowSceneIOTasks.cpp:104]: Fixed

File: src/plugins/workflow_designer/src/WorkflowViewController.cpp (hunk 2 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/workflow_designer/src/WorkflowViewController.cpp]: Fixed

**Review by atiunov — APPROVED** [review]

**Pull request merged**